test-result-summary-client/src/Build/PossibleIssues.jsx

```diff
@@ -6,45 +6,81 @@ import { getParams } from '../utils/query';
 import './table.css';
 
 export default class PossibleIssues extends Component {
+    state = {
+        error: null,
+        isLoaded: false,
+        issueTitles: [],
+        issueLinks: [],
+    };
 
-    render() {
-        const { buildId, buildName, testId, testName } = getParams( this.props.location.search );
-        
-        let issueRepos = [];
+    async componentDidMount() {
+        await this.fetchIssues();
+    }
+
+    async fetchIssues() {
+        const { buildName, testName } = getParams( this.props.location.search );
+
+        let additionalRepo = "";
         if (buildName.includes('j9') || buildName.includes('ibm')) {
-            issueRepos.push("eclipse/openj9");
+            additionalRepo = "+repo:eclipse/openj9";
         }
-        issueRepos.push("AdoptOpenJDK/openjdk-tests");
-        issueRepos.push("AdoptOpenJDK/openjdk-build");
-        issueRepos.push("AdoptOpenJDK/openjdk-infrastructure");
-
-        let issueUrls = [];
-        for (let repo of issueRepos) {
-            issueUrls.push(`https://github.com/${repo}/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+${testName}`);
-        };
-        
-        const columns = [{
-            title: 'Possible Issues',
-            dataIndex: 'issues',
-            key: 'issues',
-        }];
-
-        let dataSource = [];
-        for (let index = 0; index < issueRepos.length; index++) {
-            dataSource.push({
-                key: index,
-                issues: <a href={issueUrls[index]} target="_blank" rel="noopener noreferrer">{issueRepos[index]}</a> 
-            }); 
+
+        const response = await fetch(`https://api.github.com/search/issues?q=${testName}+state:open+repo:AdoptOpenJDK/openjdk-tests` +
+                    `+repo:AdoptOpenJDK/openjdk-infrastructure+repo:AdoptOpenJDK/openjdk-build${additionalRepo}`, {
+            method: 'get'
+        });
+        if (response.ok) {
+            const relatedIssues = await response.json();
+            let issueTitles = [];
+            let issueUrls = [];
+            for (let issue of relatedIssues.items) {
+                issueTitles.push(issue.title);
+                issueUrls.push(issue.html_url);
+            };
```

Comment:
- Can we construct `dataSource ` here, so we do not need to loop it again in render()?
- also, instead of storing isLoaded, issueTitles and issueUrls, only `dataSource` needs to be stored in state.